packages/ra-ui-materialui/src/form/SimpleFormIterator.js

```diff
@@ -64,177 +63,171 @@ const styles = theme =>
         leftIcon: {
             marginRight: theme.spacing(1),
         },
-    });
+    }),
+    { name: 'RaSimpleFormIterator' }
+);
 
-export class SimpleFormIterator extends Component {
-    constructor(props) {
-        super(props);
-        // we need a unique id for each field for a proper enter/exit animation
-        // but redux-form doesn't provide one (cf https://github.com/erikras/redux-form/issues/2735)
-        // so we keep an internal map between the field position and an auto-increment id
-        this.nextId = props.fields.length
-            ? props.fields.length
-            : props.defaultValue
-            ? props.defaultValue.length
+const SimpleFormIterator = ({
+    basePath,
+    classes: classesOverride,
+    children,
+    fields,
+    meta: { error, submitFailed },
+    record,
+    resource,
+    source,
+    disableAdd,
+    disableRemove,
+    variant,
+    margin,
+    TransitionProps,
+    defaultValue,
+}) => {
+    const translate = useTranslate();
+    const classes = useStyles({ classes: classesOverride });
+
+    // we need a unique id for each field for a proper enter/exit animation
+    // but redux-form doesn't provide one (cf https://github.com/erikras/redux-form/issues/2735)
+    // so we keep an internal map between the field position and an auto-increment id
+    let nextId =
```

Comment:
I don't understand how your code can be working. 

The right way to port a component property from this to a function component is to use a ref, otherwise the `nextId` and `ids` variables will be reset at each render.

Reply:
Sorry I couldn't finish yesterday.